exit/test_exit.c

```diff
@@ -0,0 +1,528 @@
+{
+	/* Test if parent _exit affect child process */
+	pid_t pid;
+	int pipefd[2];
+
+	/* Pipe needed for communication between grandparent and grandchild (can't use asserts in child) */
+	TEST_ASSERT_EQUAL_INT(0, pipe(pipefd));
+
+	pid = fork();
+	TEST_ASSERT_GREATER_OR_EQUAL(0, pid);
+	if (pid == 0) {
+		pid = fork();
+		/* grandchild */
+		if (pid == 0) {
+			/* Grandchild doesn't need pipe */
+			close(pipefd[0]);
+			close(pipefd[1]);
+
+			/* Sleep for a while to ensure that parent already exited */
+			sleep(1);
+			creat(PATH3, DEFFILEMODE);
+			_exit(EXIT_SUCCESS);
+		}
+		/* parent */
+		else {
+			/* Write fork return value from parent to grandparent */
+			close(pipefd[0]);
+			write(pipefd[1], &pid, sizeof(pid_t));
+			close(pipefd[1]);
+			/* parent exits right away */
+			_exit(EXIT_SUCCESS);
+		}
+	}
+	/* grandparent */
+	else {
+		int status;
+
+		close(pipefd[1]);
+		wait(&status);
+		TEST_ASSERT_EQUAL_INT(EXIT_SUCCESS, WEXITSTATUS(status));
+
+		read(pipefd[0], &pid, sizeof(pid_t));
+		TEST_ASSERT_NOT_EQUAL_INT(-1, pid);
+		close(pipefd[0]);
+
+		/* Check if file exists, if so child sleep was interrupted */
+		TEST_ASSERT_EQUAL_INT(-1, access(PATH3, F_OK));
+
+		sleep(2);
+		/* At this moment if child wasn't affected file should exist */
+		TEST_ASSERT_EQUAL_INT(0, access(PATH3, F_OK));
+
+		remove(PATH3);
+	}
+}
+
+
+TEST(unistd_exit, new_parent_id)
+{
+	/* Test that child acquire new parent ID */
+	pid_t pid;
+	int pipefd[2];
+
+	/* Pipe needed for communication between grandparent and grandchild (can't use asserts in child) */
+	TEST_ASSERT_EQUAL_INT(0, pipe(pipefd));
+	pid = fork();
+	TEST_ASSERT_GREATER_OR_EQUAL(0, pid);
+	if (pid == 0) {
+		pid = fork();
+		/* grandchild */
+		if (pid == 0) {
+			pid_t old_ppid, new_ppid;
+
+			close(pipefd[0]);
+			old_ppid = getppid();
+			/* Sleep until parents exits */
+			sleep(2);
+			/* Parent process id should be 1 (init pid) */
+			new_ppid = getppid();
+			write(pipefd[1], &old_ppid, sizeof(pid_t));
+			write(pipefd[1], &new_ppid, sizeof(pid_t));
+			close(pipefd[1]);
+			_exit(EXIT_SUCCESS);
+		}
+		/* parent */
+		else {
+			/* Write fork return value from parent to grandparent */
+			close(pipefd[0]);
+			write(pipefd[1], &pid, sizeof(pid_t));
+			close(pipefd[1]);
+			/* Sleep until child gets his parent ID */
+			sleep(1);
+			_exit(EXIT_SUCCESS);
+		}
+	}
+	/* grandparent */
+	else {
+		pid_t old_ppid, new_ppid;
+		int status;
+
+		close(pipefd[1]);
+		wait(&status);
+		TEST_ASSERT_EQUAL_INT(EXIT_SUCCESS, WEXITSTATUS(status));
+		sleep(3);
+		read(pipefd[0], &pid, sizeof(pid_t));
+		TEST_ASSERT_NOT_EQUAL_INT(-1, pid);
+		read(pipefd[0], &old_ppid, sizeof(pid_t));
+		read(pipefd[0], &new_ppid, sizeof(pid_t));
+		TEST_ASSERT_NOT_EQUAL_INT(old_ppid, new_ppid);
+		close(pipefd[0]);
+	}
+}
+
+
+TEST(unistd_exit, SIGCHLD_sent)
+{
+	/* Test that SIGCHILD signal is sent after child exits */
+	pid_t pid;
+	int status, ret;
+
+	struct sigaction sa;
+	sa.sa_handler = test_handler;
```

Comment:
Did you want to test this paragraph? If so, handler should be SIG_IGN:
```suggestion
	sa.sa_handler = SIG_IGN;
```
![Screenshot from 2023-08-10 18-40-40](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/77304431/863978b9-8036-43fe-b957-fb3efd997f4d)

Reply:
![image](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/108683715/c3053b9f-b1d4-42a4-9856-1ef63cf94be1)